lib/svtplay_dl/utils/retriever_multi.py

```diff
@@ -0,0 +1,111 @@
+# ex:ts=4:sw=4:sts=4:et
+# -*- tab-width: 4; c-basic-offset: 4; indent-tabs-mode: nil -*-
+
+import pycurl
+from io import BytesIO
+import time
+from datetime import timedelta
+
+# We should ignore SIGPIPE when using pycurl.NOSIGNAL - see
+# the libcurl tutorial for more info.
+try:
+    import signal
+    from signal import SIGPIPE, SIG_IGN
+except ImportError:
+    pass
+else:
+    signal.signal(SIGPIPE, SIG_IGN)
+
+from svtplay_dl.utils.output import progressbar, progress_stream, ETA, output
+
+FIREFOX_UA = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/45.0.2454.101 Safari/537.3'
+DEBUG = False
+
+
+def get(urls, cookies, silent=False, live=False, total_duration=None, tot_urls=0, num_conn=100):
+
+    # Check args
+    num_urls = len(urls)
+    num_conn = min(num_conn, num_urls)
+    eta = ETA(num_urls)
+
+    if DEBUG:
+        print("PycURL %s (compiled against 0x%x)" % (pycurl.version, pycurl.COMPILE_LIBCURL_VERSION_NUM))
+        print("----- Getting", num_urls, "URLs using", num_conn, "connections -----")
+
+    start_time = time.time()
+
+    # Pre-allocate a list of curl objects
+    m = pycurl.CurlMulti()
+    m.handles = []
+    for i in range(num_conn):
+        c = pycurl.Curl()
+        c.b = None
+        c.setopt(pycurl.FOLLOWLOCATION, 1)
+        c.setopt(pycurl.MAXREDIRS, 5)
+        c.setopt(pycurl.CONNECTTIMEOUT, 30)
+        c.setopt(pycurl.TIMEOUT, 300)
+        c.setopt(pycurl.NOSIGNAL, 1)
+        c.setopt(pycurl.HTTPHEADER, ["User-Agent:" + FIREFOX_UA])
+        m.handles.append(c)
+
+    freelist = m.handles[:]
+    num_processed = 0
+    last_num_processed = 0
+    while num_processed < num_urls:
+        # If there is an url to process and a free curl object, add to multi stack
+        while urls and freelist:
+            url = urls.pop(0)
+            c = freelist.pop()
+            c.b = BytesIO()
+            #c.setopt(pycurl.COOKIE, cookies)
```

Comment:
uncomment this?

Reply:
Do you have a url that needs cookies?

Reply:
Reading the hls code, i saw that i added the cookies for Dplay back in 2015. but we dont need it today. 
is there a reason this is comment out?

Reply:
I didn't get it to work

Reply:
So, should i remove it?

Reply:
this seems to work:
```
In [1]: import pycurl

In [2]: c = pycurl.Curl()

In [3]: c.setopt(c.URL, "https://httpbin.org/cookies")

In [4]: c.setopt(pycurl.COOKIE, "kalle2=kek;anka=2k2")

In [5]: c.perform()
{
  "cookies": {
    "anka": "2k2",
    "kalle2": "kek"
  }
}
```
Need to translate the cookie dict to `key1=value1;key2=value2` with `;` between the keys.

Reply:
Done